src/com/nisovin/magicspells/castmodifiers/conditions/UnderBlockCondition.java

```diff
@@ -0,0 +1,107 @@
+package com.nisovin.magicspells.castmodifiers.conditions;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+
+import com.nisovin.magicspells.materials.MagicMaterial;
+import com.nisovin.magicspells.util.RegexUtil;
+import com.nisovin.magicspells.DebugHandler;
+import com.nisovin.magicspells.MagicSpells;
+import com.nisovin.magicspells.castmodifiers.Condition;
+
+public class UnderBlockCondition extends Condition {
+
+	int height;
+	String blocks;
+	Set<Material> types;
+	List<MagicMaterial> mats;
+	MagicMaterial mat;
```

Comment:
Redundant with the list on line 26, no significant benefit of using a single instance special case.